shell/platform/android/platform_view_android_jni_impl.cc

```diff
@@ -145,6 +146,31 @@ static void DestroyJNI(JNIEnv* env, jobject jcaller, jlong shell_holder) {
   delete ANDROID_SHELL_HOLDER;
 }
 
+// Signature is similar to RunBundleAndSnapshotFromLibrary but it can't change
+// the bundle path or asset manager since we can only spawn with the same
+// AOT.
+static jlong SpawnJNI(JNIEnv* env,
+                      jobject jcaller,
+                      jobject flutterJNI,
+                      jlong shell_holder,
+                      jstring jEntrypoint,
+                      jstring jLibraryUrl) {
+  fml::jni::JavaObjectWeakGlobalRef java_jni(env, flutterJNI);
+  std::shared_ptr<PlatformViewAndroidJNI> jni_facade =
+      std::make_shared<PlatformViewAndroidJNIImpl>(java_jni);
+
+  auto entrypoint = fml::jni::JavaStringToString(env, jEntrypoint);
+  auto libraryUrl = fml::jni::JavaStringToString(env, jLibraryUrl);
+
+  auto spawned_shell_holder =
+      ANDROID_SHELL_HOLDER->Spawn(jni_facade, entrypoint, libraryUrl);
+  if (spawned_shell_holder->IsValid()) {
+    return reinterpret_cast<jlong>(spawned_shell_holder.release());
+  } else {
+    return 0;
```

Comment:
Would it be more appropriate to throw a java exception?

Reply:
I think so, but it's it should be consistent with the mainline path nativeAttach. It's a pretty big breaking change to make it conform there. Instead, the current pattern is defensive coding on the Java side. Doing that as well here.